spec/features/signin_process_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'the signin process', type: :feature do
+  subject do
+    visit login_path
+    click_on 'login'
+  end
+
+  context 'when API Gouv authentication is successful' do
+    before do
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:api_gouv] = OmniAuth::AuthHash.new({
+        info: {
+          email: user.email,
+          sub: user.oauth_api_gouv_id,
+        }
+      })
+    end
+
+    after { OmniAuth.config.test_mode = false }
+
+    context 'when the user is unknown from API Entreprise' do
+      let!(:user) { build(:user) }
+
+      it 'redirects to the login page' do
+        subject
+
+        expect(page.current_path).to eq(login_path)
+      end
+    end
+
+    context 'when the user is an admin' do
+      let!(:user) { create(:user, :admin) }
+
+      it 'redirects to the users index page' do
+        subject
+
+        expect(page.current_path).to eq(admin_users_path)
+      end
+    end
+
+    context 'when the user is not an admin' do
+      let!(:user) { create(:user) }
+
+      it 'redirects to the user detail page' do
+        subject
+
+        expect(page.current_path).to eq(user_path(user))
+      end
+    end
+  end
+
+  context 'when API Gouv authentication fails' do
+    before do
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:api_gouv] = :invalid_credentials
+    end
+
+    after { OmniAuth.config.test_mode = false }
+
+    it 'redirects to the login page' do
+      subject
+
+      expect(page.current_path).to eq(login_path)
+    end
+
+    it 'displays an error message' do
+      subject
+
+      expect(find('div.notice').text).to eq('Invalid credentials')
```

Comment:
```suggestion
      expect(page).to have_content('Invalid credentials')
```

idem, je me ferai pas chier je matcherai juste le `.notice` 🤷

D'ailleurs je mettrai carrément le code technique plutôt que humanized mais c'est du détail.

Reply:
Je pense qu'il y aura une passe sur les messages d'erreurs à destinations des end users à terme, dans lesquels on inclura les codes techniques. 

Je vais retirer le `.humanize` au profit du code, c'est uniquement pour nous de toute façon dans un premier temps.